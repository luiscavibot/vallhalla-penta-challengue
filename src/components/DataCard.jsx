import React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import Rating from '@mui/material/Rating';
import PropTypes from 'prop-types';

export const DataCard = ({ movie }) => {
	return (
		<Card
			sx={{
				height: '100%',
			}}>
			<CardMedia component="img" image={movie.image} alt={movie.title} />
			<CardContent>
				<Typography gutterBottom variant="h5" component="div">
					{movie.title}
				</Typography>

				<Typography variant="subtitle2">{movie.crew}</Typography>
				<Typography variant="body2" color="text.secondary">
					Crew
				</Typography>

				<Typography variant="subtitle2">{movie.year}</Typography>
				<Typography variant="body2" color="text.secondary">
					Year
				</Typography>

				<Typography variant="subtitle2">{movie.rank}</Typography>
				<Typography variant="body2" color="text.secondary">
					Rank
				</Typography>

				<Rating
					size="small"
					name="read-only"
					value={movie.imDbRating * 0.5}
					readOnly
					precision={0.2}
				/>
				<Typography variant="body2" color="text.secondary">
					ImDb Rating
				</Typography>
			</CardContent>
		</Card>
	);
};
DataCard.propTypes = {
	movie: PropTypes.object.isRequired,
};
