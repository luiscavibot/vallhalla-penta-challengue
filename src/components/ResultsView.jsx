import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@mui/material';
import { DataCard } from './DataCard';
import { sliceMoviesArray } from '../helpers/SliceMoviesArray';

export const ResultsView = ({ movies, page }) => {
	let moviesSlice = sliceMoviesArray(movies, page);
	return (
		<Grid container spacing={2}>
			{moviesSlice.map((movie) => (
				<Grid item xs={12} sm={6} md={3} key={movie.id}>
					<DataCard movie={movie} />
				</Grid>
			))}
		</Grid>
	);
};

export default ResultsView;
ResultsView.propTypes = {
	movies: PropTypes.array.isRequired,
	page: PropTypes.number.isRequired,
};
