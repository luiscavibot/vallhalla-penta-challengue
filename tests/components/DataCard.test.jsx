import { render, screen } from '@testing-library/react';
import { DataCard } from '../../src/components/DataCard';

describe('Pruebas en <DataCard />', () => {
	const movie = {
		id: 'tt0111161',
		rank: '1',
		title: 'The Shawshank Redemption',
		fullTitle: 'The Shawshank Redemption (1994)',
		year: '1994',
		image: 'https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_UX128_CR0,3,128,176_AL_.jpg',
		crew: 'Frank Darabont (dir.), Tim Robbins, Morgan Freeman',
		imDbRating: '9.2',
		imDbRatingCount: '2608426',
	};
	test('Debe hacer match con el snapshot', () => {
		const { container } = render(<DataCard movie={movie} />);
		expect(container).toMatchSnapshot();
	});
	test('Debe tener una etiqueta img con un src igual a image y un alt igual al title del objeto', () => {
		render(<DataCard movie={movie} />);
		const { src, alt } = screen.getByAltText(movie.title);
		expect(src).toBe(movie.image);
		expect(alt).toBe(movie.title);
	});
});
