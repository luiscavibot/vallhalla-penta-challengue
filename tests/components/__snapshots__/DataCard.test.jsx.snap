// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DataCard /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="The Shawshank Redemption"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_UX128_CR0,3,128,176_AL_.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        The Shawshank Redemption
      </div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Frank Darabont (dir.), Tim Robbins, Morgan Freeman
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Crew
      </p>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        1994
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Year
      </p>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        1
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Rank
      </p>
      <span
        aria-label="4.6 Stars"
        class="MuiRating-root MuiRating-sizeSmall MuiRating-readyOnly css-1lauo1g-MuiRating-root"
        role="img"
      >
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 60.00000000000001%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        ImDb Rating
      </p>
    </div>
  </div>
</div>
`;
